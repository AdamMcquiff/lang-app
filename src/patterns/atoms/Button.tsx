import * as React from 'react';
import './Button.scss';

interface IButtonProps {
    a11yLabel: string,
    textLabel: string,
    onClick(): void
}

class Button extends React.Component<IButtonProps> {
    public static defaultProps = {
        a11yLabel: 'Click me.',
        textLabel: 'Click me.'
    }

    public render() {
        return (
            <button className="button" 
                    aria-label={this.props.a11yLabel}
                    onClick={this.props.onClick}>
                {this.props.textLabel}
            </button>
        )
    }
}

export default Button;