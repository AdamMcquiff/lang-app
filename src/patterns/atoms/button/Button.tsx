import * as React from 'react';
import './Button.scss';

interface IButtonProps {
    a11yLabel: string,
    className: string,
    textLabel: string,
    onClick(): void
}

class Button extends React.Component<IButtonProps> {
    public static defaultProps = {
        a11yLabel: 'Click me',
        className: '',
        onClick: () => null,
        textLabel: 'Click me',
    }

    public render() {
        return (
            <button className={'a-button ' + this.props.className}
                    aria-label={this.props.a11yLabel}
                    onClick={this.props.onClick}>
                {this.props.textLabel}
            </button>
        )
    }
}

export default Button;