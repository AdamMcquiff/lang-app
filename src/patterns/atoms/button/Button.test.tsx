import { mount, shallow } from 'enzyme';
import * as React from 'react';
import Button from './Button';

describe('Button', () => {
    const props = {
        a11yLabel: "Some label",
        onClick: jest.fn(),
        textLabel: "Some label"
    };

    it('renders without crashing', () => {
        const component = shallow(<Button />);
        expect(component).toMatchSnapshot();
    });

    it('renders correctly with a11yLabel prop passed', () => {
        const component = mount(<Button a11yLabel={props.a11yLabel} />);
        const renderedA11yLabel = component.find('button').getDOMNode().getAttribute('aria-label');
        expect(renderedA11yLabel).toEqual(props.a11yLabel); 
    });

    it('renders correctly with textLabel prop passed', () => {
        const component = mount(<Button textLabel={props.textLabel} />);
        const renderedTextLabel = component.find('button').text();
        expect(renderedTextLabel).toEqual(props.textLabel); 
    });

    it('renders correctly with onClick prop passed', () => {
        const component = mount(<Button onClick={props.onClick} />);
        component.find('button').simulate('click');
        expect(props.onClick).toBeCalled(); 
    });
});