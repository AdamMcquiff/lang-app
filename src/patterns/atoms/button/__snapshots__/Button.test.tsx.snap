// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders without crashing 1`] = `
<button
  aria-label="Click me"
  className="a-button"
  onClick={[Function]}
>
  Click me
</button>
`;
